### PR TITLE
Add except: ["after-rule"] to rule-nested-empty-line-before

### PR DESCRIPTION
**Pull request by rontav**
> Which issue, if any, is this issue related to?

#2179 

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/checkRuleEmptyLineBefore.js [hunk 1]
```diff
@@ -23,6 +23,11 @@ module.exports = function (opts) {
     expectEmptyLineBefore = !expectEmptyLineBefore
   }
 
+  // Optionally reverse the expectation for the first nested selector
```
Comment (jeddy3) [inline lib/rules/checkRuleEmptyLineBefore.js]: "Optionally reverse the expectation if a rule precedes this node"

File: lib/rules/rule-nested-empty-line-before/README.md [hunk 2]
```diff
@@ -152,6 +152,35 @@ The following patterns are *not* considered warnings:
 }
 ```
 
+### `except: ["after-rule"]`
+
+Reverse the primary option if the rule is the first after a property.
```
Comment (jeddy3) [inline lib/rules/rule-nested-empty-line-before/README.md]: "Reverse the primary option if the rule comes after rule"

File: lib/rules/rule-nested-empty-line-before/README.md [hunk 3]
```diff
@@ -152,6 +152,35 @@ The following patterns are *not* considered warnings:
...
+
+For example, with `"never"`:
```
Comment (jeddy3) [inline lib/rules/rule-nested-empty-line-before/README.md]: "For example, with `"always"`:"

Once you switch to `opts.rule.prev().type === "rule"` you'll hopefully see why this should be `"always"` and not `"never"`

File: lib/rules/rule-nested-empty-line-before/__tests__/index.js [hunk 4]
```diff
@@ -122,6 +122,40 @@ testRule(rule, {
 
 testRule(rule, {
   ruleName,
+  config: [ "always", { except: ["after-rule"] } ],
```
Comment (jeddy3) [inline lib/rules/rule-nested-empty-line-before/__tests__/index.js]: These tests will likely fail when you update the logic.

Can you also include a test (both passing and failing) for when a rule is preceded by an at-rule e.g.

```
a {
  @media {
    b {}
  }
  
  d {}
  d {}
}
```

**Comment by rontav** [comment]
Also, I've added a test case for atrule.
At first, I didn't agree with the "backwards" logic, but I understand why is it more stable that way.

File: lib/rules/rule-nested-empty-line-before/__tests__/index.js [hunk 5]
```diff
@@ -122,6 +122,50 @@ testRule(rule, {
...
+    message: messages.expected,
+  }, {
+    code: "a {\n color: pink;\n b {color: red; }\n c {color: blue; }\n}",
+    message: messages.expected,
+  }, {
+    code: "a {\n @media {\n\n   b {}\n }\n c {}\n d {}\n}",
+    description: "media rule",
+    message: messages.expected,
+  }, {
+    code: "a {\r\n color: pink;\r\n b {\rcolor: red; \r}\r\n c {\rcolor: blue; \r}\r\n}",
```
Comment (jeddy3) [inline lib/rules/rule-nested-empty-line-before/__tests__/index.js]: There's three `\r`'s here unaccompanied by `\n`'s. Can you add the `\n`'s please.

File: lib/rules/rule-nested-empty-line-before/__tests__/index.js [hunk 6]
```diff
@@ -122,6 +122,50 @@ testRule(rule, {
...
+    description: "CRLF",
+  } ],
+
+  reject: [ {
+    code: "a {\n color: pink; b {color: red; }\n c {color: blue; }\n}",
+    message: messages.expected,
+  }, {
+    code: "a {\n color: pink;\n b {color: red; }\n c {color: blue; }\n}",
+    message: messages.expected,
+  }, {
```
Comment (jeddy3) [inline lib/rules/rule-nested-empty-line-before/__tests__/index.js]: Let's add a test here to show that empty lines before the _second nested rule_ are being checked e.g.

```js
{
code: "a {\n color: pink;\n\n b {color: red; }\n\n c {color: blue; }\n}",
message: messages.expected,
}
```

File: lib/rules/rule-nested-empty-line-before/__tests__/index.js [hunk 7]
```diff
@@ -230,6 +274,47 @@ testRule(rule, {
...
+    description: "media rule",
+  }, {
+    code: "a {\n color: pink; \r\n b {color: red; } \n\n c {color: blue; }\n}",
+    description: "CRLF",
+  } ],
+
+  reject: [ {
+    code: "a {\n color: pink;\n\n b {color: red; } \n\n c {color: blue; }\n}",
+    message: messages.rejected,
+  }, {
```
Comment (jeddy3) [inline lib/rules/rule-nested-empty-line-before/__tests__/index.js]: Let's add a test here to show that empty lines before the _second nested rule_ are being checked e.g.

```js
{
code: "a {\n color: pink;\n b {color: red; } \n c {color: blue; }\n}",
message: messages.rejected,
}
```

**Comment by jeddy3** [comment]
@rontav Excellent job. It LGTM. Another stylelint member will likely look it over before we merge though.

We'll squash the commits within github, so don't worry about doing that.

File: lib/rules/rule-nested-empty-line-before/README.md [hunk 8]
```diff
@@ -152,6 +152,35 @@ The following patterns are *not* considered warnings:
 }
 ```
 
+### `except: ["after-rule"]`
+
+Reverse the primary option if the rule comes after rule.
```
Comment (davidtheclark) [inline lib/rules/rule-nested-empty-line-before/README.md]: Let's add "another": after another rule.

**Comment by jeddy3** [comment]
@rontav Thanks again!

Changelog:

> -   Added: `except: ["after-rule"]` option to `rule-nested-empty-line-before` ([#2188](https://github.com/stylelint/stylelint/pull/2188)).
